### PR TITLE
fix: Remove date partition column field from datasources that don't s…

### PR DESCRIPTION
**Pull request by achals**
…upport them

Signed-off-by: Achal Shah <achals@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: sdk/python/feast/infra/offline_stores/bigquery_source.py [hunk 1]
```diff
@@ -61,6 +60,14 @@ def __init__(
             table = table_ref
         self.bigquery_options = BigQueryOptions(table_ref=table, query=query)
 
+        if date_partition_column:
+            warnings.warn(
```
Comment (felixwang9817) [inline sdk/python/feast/infra/offline_stores/bigquery_source.py:64]: do we want to add a specific version warning? e.g. Feast 0.21+ onwards will not support

and same for other sources

File: sdk/python/feast/infra/offline_stores/file_source.py [hunk 2]
```diff
@@ -114,7 +121,6 @@ def from_proto(data_source: DataSourceProto):
             path=data_source.file_options.uri,
             event_timestamp_column=data_source.event_timestamp_column,
             created_timestamp_column=data_source.created_timestamp_column,
-            date_partition_column=data_source.date_partition_column,
```
Comment (felixwang9817) [inline sdk/python/feast/infra/offline_stores/file_source.py:117]: I think you can make this same change for all the other sources

File: sdk/python/tests/integration/feature_repos/universal/data_sources/file.py [hunk 3]
```diff
@@ -48,7 +48,6 @@ def create_data_source(
             path=f"{f.name}",
             event_timestamp_column=event_timestamp_column,
             created_timestamp_column=created_timestamp_column,
-            date_partition_column="",
```
Comment (felixwang9817) [inline sdk/python/tests/integration/feature_repos/universal/data_sources/file.py:51]: you also missed line 133

File: sdk/python/tests/integration/registration/test_registry.py [hunk 4]
```diff
@@ -169,7 +169,6 @@ def test_apply_feature_view_success(test_registry):
         path="file://feast/*",
         event_timestamp_column="ts_col",
         created_timestamp_column="timestamp",
-        date_partition_column="date_partition_col",
```
Comment (felixwang9817) [inline sdk/python/tests/integration/registration/test_registry.py:172]: you missed a few more in this file

**Review by felixwang9817 — CHANGES_REQUESTED** [review]
btw I think you should also add the warning in `DataSource.__init__`

File: sdk/python/feast/infra/offline_stores/bigquery_source.py [hunk 5]
```diff
@@ -20,8 +20,8 @@ def __init__(
         table: Optional[str] = None,
         table_ref: Optional[str] = None,
         created_timestamp_column: Optional[str] = "",
+        date_partition_column: Optional[str] = None,
```
[the remark this answers is not in the document]
Comment (achals) [reply sdk/python/feast/infra/offline_stores/bigquery_source.py:23]: oops

File: sdk/python/feast/infra/offline_stores/bigquery_source.py (hunk 1 above)
Comment (achals) [reply sdk/python/feast/infra/offline_stores/bigquery_source.py:64]: Yeah I'll add that

File: sdk/python/feast/infra/offline_stores/file_source.py (hunk 2 above)
Comment (achals) [reply sdk/python/feast/infra/offline_stores/file_source.py:117]: I *think* I have? Assuming you're referring to not passing `date_partition_column` into the constructor here?

File: sdk/python/tests/integration/feature_repos/universal/data_sources/file.py (hunk 3 above)
Comment (achals) [reply sdk/python/tests/integration/feature_repos/universal/data_sources/file.py:51]: good catch

File: sdk/python/tests/integration/registration/test_registry.py (hunk 4 above)
Comment (achals) [reply sdk/python/tests/integration/registration/test_registry.py:172]: I'll fix em

File: sdk/python/feast/infra/offline_stores/bigquery_source.py [hunk 6]
```diff
@@ -14,14 +14,18 @@
 
 
 class BigQuerySource(DataSource):
+    # Note: Python requires redefining hash in child classes that override __eq__
```
Comment (felixwang9817) [inline sdk/python/feast/infra/offline_stores/bigquery_source.py:17]: why did this get moved?

File: sdk/python/feast/infra/offline_stores/bigquery_source.py [hunk 7]
```diff
@@ -78,19 +91,14 @@ def __init__(
 
         super().__init__(
             _name if _name else "",
-            event_timestamp_column,
-            created_timestamp_column,
-            field_mapping,
-            date_partition_column,
```
Comment (felixwang9817) [inline sdk/python/feast/infra/offline_stores/bigquery_source.py:84]: btw you can remove `date_partition_column` in `to_proto` and `from_proto`

File: sdk/python/feast/infra/offline_stores/file_source.py (hunk 2 above)
Comment (felixwang9817) [reply sdk/python/feast/infra/offline_stores/file_source.py:117]: ah I meant remove from `from_proto` and `to_proto`, e.g. I don't think you did that for `bigquery_source.py`

**Review by felixwang9817 — CHANGES_REQUESTED** [review]
couple more nits

File: sdk/python/feast/infra/offline_stores/bigquery_source.py (hunk 6 above)
Comment (achals) [reply sdk/python/feast/infra/offline_stores/bigquery_source.py:17]: formatter 🤷🏽

File: sdk/python/feast/infra/offline_stores/bigquery_source.py (hunk 7 above)
Comment (achals) [reply sdk/python/feast/infra/offline_stores/bigquery_source.py:84]: done

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Pull request merged**